chrome/content/zotero/integration/quickFormat.js

```diff
@@ -1367,23 +1550,211 @@ var Zotero_QuickFormat = new function () {
 				spinner.removeAttribute("status");
 			});
 	}
+
+	function lastBubbleBeforePoint(x, y) {
```

Comment:
Also can we not do this by checking the elements being dragged over? If not, can we not reuse some of the code used in item tree dragging, particularly what is available in components/utils.js? Also, if this is really necessary, can you add some documentation?

Reply:
`getLastBubbleBeforePoint`'s initial and main function is to determine after which bubble to place an empty input on click between bubbles. I added some comments to make it a bit clearer what's going on there.

I used it again in `dragover` listener because it allows us to handle an edge case. If a bubble is dragged over another bubble, that other bubble is used as a pivot point from before/after which placeholder lands. But if a bubble is dragged to an empty space in the editor without passing over bubbles on the way, the functionality of finding the last bubble before a coordinate tells us where to insert the placeholder. That situation is  possible, for example if the editor has 2 lines of bubbles, the second line has only 1 or 2 bubbles and we drag the last bubble from the first line to the empty space on the second line.

I see there is `getDragTargetOrient` in `components/utils.js` that does pretty much exactly what we want here but on the Y-axis, not the X-axis (which makes sense for the item tree). Correct me if there's something I missed there